pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook.go

```diff
@@ -111,18 +111,14 @@ func (v *fleetResourceValidator) handleMemberCluster(req admission.Request) admi
 
 // handleInternalMemberCluster allows/denies the request to modify internal member cluster object after validation.
 func (v *fleetResourceValidator) handleInternalMemberCluster(ctx context.Context, req admission.Request) admission.Response {
-	var currentIMC fleetv1alpha1.InternalMemberCluster
-	if err := v.decodeRequestObject(req, &currentIMC); err != nil {
+	var imc fleetv1alpha1.InternalMemberCluster
+	if err := v.decodeRequestObject(req, &imc); err != nil {
 		return admission.Errored(http.StatusBadRequest, err)
 	}
 	if req.Operation == admissionv1.Update {
-		var oldIMC fleetv1alpha1.InternalMemberCluster
-		if err := v.decoder.DecodeRaw(req.OldObject, &oldIMC); err != nil {
-			return admission.Errored(http.StatusBadRequest, err)
-		}
-		return validation.ValidateInternalMemberClusterUpdate(ctx, v.client, currentIMC, oldIMC, v.whiteListedUsers, req.UserInfo)
+		return validation.ValidateInternalMemberClusterUpdate(ctx, v.client, imc, v.whiteListedUsers, req.UserInfo, req.SubResource)
```

Comment:
logic wise, it seems to be cleaner to only handle the status update here (with a function) and leave the rest to common resources.